crates/proving-service-client/build.rs

```diff
@@ -1,81 +1,61 @@
-}
+    let proving_service_descriptor = proving_service_api_descriptor();
 
-// HELPER FUNCTIONS
-// ================================================================================================
+    // Build std version
+    let std_path = format!("{GENERATED_OUT_DIR}/std");
+    build_tonic_from_descriptor(proving_service_descriptor.clone(), std_path, true)?;
 
-/// Copies the proto file from the root proto directory to the proto directory of this crate.
-fn copy_proto_files() -> miette::Result<()> {
-    let src_file = format!("{REPO_PROTO_DIR}/proving_service.proto");
-    let dest_file = format!("{CRATE_PROTO_DIR}/proving_service.proto");
+    // Build nostd version
+    let nostd_path = format!("{GENERATED_OUT_DIR}/nostd");
+    build_tonic_from_descriptor(proving_service_descriptor, nostd_path.clone(), false)?;
 
-    fs::remove_dir_all(CRATE_PROTO_DIR).into_diagnostic()?;
-    fs::create_dir_all(CRATE_PROTO_DIR).into_diagnostic()?;
-    fs::copy(src_file, dest_file).into_diagnostic()?;
+    // Convert nostd version to use core/alloc instead of std
+    let nostd_file_path = format!("{nostd_path}/proving_service.rs");
+    convert_to_nostd(&nostd_file_path)?;
 
     Ok(())
 }
 
-fn compile_tonic_client_proto() -> miette::Result<()> {
-    let crate_root =
-        PathBuf::from(env::var("CARGO_MANIFEST_DIR").expect("CARGO_MANIFEST_DIR should be set"));
-    let dst_dir = crate_root.join("src").join("proving_service").join("generated");
-
-    // Remove `proving_service.rs` if it exists.
-    // We don't need to check the success of this operation because the file may not exist.
-    let _ = fs::remove_file(dst_dir.join("std").join("proving_service.rs"));
-    let _ = fs::remove_file(dst_dir.join("nostd").join("proving_service.rs"));
-
-    let out_dir = env::var("OUT_DIR").into_diagnostic()?;
-    let file_descriptor_path = PathBuf::from(out_dir).join("file_descriptor_set.bin");
-
-    let proto_dir: PathBuf = CRATE_PROTO_DIR.into();
-    let protos = &[proto_dir.join("proving_service.proto")];
-    let includes = &[proto_dir];
-
-    let file_descriptors = protox::compile(protos, includes)?;
-    fs::write(&file_descriptor_path, file_descriptors.encode_to_vec()).into_diagnostic()?;
+// HELPER FUNCTIONS
+// ================================================================================================
 
-    // Codegen for wasm transport and std transport
-    let nostd_path = dst_dir.join("nostd");
-    let std_path = dst_dir.join("std");
-    build_tonic_client(&file_descriptor_path, &std_path, protos, includes, false)?;
-    build_tonic_client(&file_descriptor_path, &nostd_path, protos, includes, true)?;
+/// Builds tonic code from a `FileDescriptorSet` with specified configuration
+fn build_tonic_from_descriptor(
+    descriptor: FileDescriptorSet,
+    out_dir: String,
+    build_transport: bool,
+) -> miette::Result<()> {
+    tonic_build::configure()
+        .out_dir(out_dir)
+        .build_server(false)
```

Comment:
I think this setting disabled building of the server files, right?

Reply:
Yes, it does disables the server generation.